include/nil/crypto3/zk/commitments/polynomial/kzg_v2.hpp

```diff
@@ -0,0 +1,348 @@
+                private:
+                    params_type _params;
+                    std::map<std::size_t, commitment_type> _commitments;
+                    std::map<std::size_t, std::vector<typename KZGScheme::single_commitment_type>> _ind_commitments;
+                    std::vector<typename KZGScheme::scalar_value_type> _merged_points;
+                protected:
+
+                    // Differs from static one by input parameters
+                    void merge_eval_points() {
+                        std::set<typename KZGScheme::scalar_value_type> set;
+                        for( auto const &it:this->_points){
+                            auto k = it.first;
+                            for (std::size_t i = 0; i < this->_points[k].size(); ++i) {
+                                set.insert(this->_points[k][i].begin(), this->_points[k][i].end());
+                            }
+                        }
+                        _merged_points = std::vector<typename  KZGScheme::scalar_value_type>(set.begin(), set.end());
+                    }
+
+                    typename math::polynomial<typename KZGScheme::scalar_value_type>
+                    set_difference_polynom(
+                        std::vector<typename KZGScheme::scalar_value_type> merged_points,
+                        std::vector<typename KZGScheme::scalar_value_type> points
+                    ) {
+                        std::sort(merged_points.begin(), merged_points.end());
+                        std::sort(points.begin(), points.end());
+                        std::vector<typename KZGScheme::scalar_value_type> result;
+                        std::set_difference(merged_points.begin(), merged_points.end(), points.begin(), points.end(), std::back_inserter(result));
+                        if (result.size() == 0) {
+                            return typename math::polynomial<typename KZGScheme::scalar_value_type>({{1}});
+                        }
+                        BOOST_ASSERT(this->get_V(result) * this->get_V(points) == this->get_V(merged_points));
+                        return this->get_V(result);
+                    }
+
+                    void update_transcript(std::size_t batch_ind, typename KZGScheme::transcript_type &transcript) {
+                        /* The procedure of updating the transcript is subject to review and change
+                         * #295 */
+
+                        // Push commitments to transcript
+                        transcript(_commitments[batch_ind]);
+
+                        // Push evaluation points to transcript
+                        for( std::size_t i = 0; i < this->_z.get_batch_size(batch_ind); i++){
+                            for( std::size_t j = 0; j < this->_z.get_poly_points_number(batch_ind, i); j++  ) {
+                                nil::marshalling::status_type status;
+                                std::vector<uint8_t> byteblob =
```

Comment:
It's unrelated to this MR, but please make sure that you change this code in your Poseidon changes.